lldb/tools/lldb-dap/Handler/EvaluateRequestHandler.cpp

```diff
@@ -181,7 +181,17 @@ void EvaluateRequestHandler::operator()(
         expression = dap.last_nonempty_var_expression;
       else
         dap.last_nonempty_var_expression = expression;
+    } else {
+      // If this isn't a REPL context, trim leading pointer/reference characters
+      // to ensure we return the actual value of the expression.
```

Comment:
doesn't this also trigger in way too many other contexts? E.g., also in the watch panel? I think in the watch panel a `*x` should actually dereference `x`

Reply:
Since people have to voluntarily put something in a watch panel, then in that case we should even run it as an expression.  If somebody actually puts `i++` in a watch window, presumably they meant for it to increment every time they stepped.
We had a silly bug in gdbtk way back in the day where we if you moused over a selection we would run the selection  as an expression.  But lots of people seem to select expressions so that they can focus on just the part they selected.  As you can imagine, that went poorly.
The main thing - for me - is not to always have running expressions be a distinct gesture.

Reply:
I agree that we shouldn't run evaluate expressions like `i++` in a watch context.

I can fix that as well in this change as well. We're trying the expression with `frame.EvaluateExpression` if its not 'hover' but that should really be only used if 'context' == 'repl'.

I also updated this to only trim leading `&*` for 'hover' specifically since watching `*x` should be supported.